src/ScnSocialAuth/Authentication/Adapter/HybridAuth.php

```diff
@@ -154,6 +154,7 @@ public function authenticate(AuthEvent $authEvent)
             $mapper = $this->getZfcUserMapper();
             $user = $mapper->findById($localUserProvider->getUserId());
             if (!in_array($user->getState(), $zfcUserOptions->getAllowedLoginStates())) {
+                $this->hybridAuth->getAdapter($provider)->adapter->setUserUnconnected();
```

Comment:
It might work, I find this way much clearer but I found out that the "getAdapter" function call does not cache adapters by providers (https://github.com/hybridauth/hybridauth/blob/master/hybridauth/Hybrid/Auth.php) so it would be more efficient to use the $adapter from L81 if it's possible.

I'll test this solution ASAP.